src/main/java/com/google/vinet/data/IsolateTimeSlot.java

```diff
@@ -16,15 +16,62 @@
 
 package com.google.vinet.data;
 
+import com.google.appengine.api.datastore.*;
 import java.time.Instant;
+import java.time.LocalDate;
 
-public class IsolateTimeSlot extends TimeSlot {
+public class IsolateTimeSlot extends TimeSlot implements Datastoreable{
+  protected final static String ISOLATE_TIME_SLOT_TABLE_NAME = "IsolateTimeSlot";
 
-  protected IsolateTimeSlot(Instant start, Instant end, Isolate isolate) {
+  protected final Key ticket;
+  protected final LocalDate date;
+  protected final DatastoreService datastore;
+
+  /**
+   * Construct an IsolateTimeSlot with a dependency on the default implementation of Datastore.
+   * @param start The start of the TimeSlot.
+   * @param end The end of the TimeSlot.
+   * @param isolate The isolate associated with the TimeSlot.
+   * @param date The date on which the TimeSlot is scheduled.
+   * @param ticket The Datastore key of the request ticket for this TimeSlot.
+   */
+  public IsolateTimeSlot(Instant start, Instant end, Isolate isolate, LocalDate date, Key ticket) {
+    this(start, end, isolate, date, ticket, DatastoreServiceFactory.getDatastoreService());
+  }
+
+  /**
+   Construct an IsolateTimeSlot with a dependency on the provided implementation of Datastore.
+   * @param start The start of the TimeSlot.
+   * @param end The end of the TimeSlot.
+   * @param isolate The isolate associated with the TimeSlot.
+   * @param date The date on which the TimeSlot is scheduled.
+   * @param ticket The Datastore key of the request ticket for this TimeSlot.
+   * @param datastore The implementation of Datastore to depend on.
+   */
+  public IsolateTimeSlot(Instant start, Instant end, Isolate isolate, LocalDate date, Key ticket, DatastoreService datastore) {
```

Comment:
I had wanted to continue supporting the pre-existing api, with no datastore injection. An earlier iteration of the class hadn't allowed for dependency injection, so in order to prevent pusing a breaking change, dependency injection was made optional.